### PR TITLE
make sure attribute descriptors are honored to map json to xml

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

See #134 

#### Related issues

- connect to <link_to_referenced_issue>

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by raymondfeng** [comment]
@bhavnagyanchandani, Please verify if this patch works for you.

**Comment by bhavnagyanchandani** [comment]
@raymondfeng - I have tested the code from the branch and it is working perfectly fine now and the issue seems to be resolved. Thanks for helping to fix the issue.
I would be looking forward to get this branch merged to master.  cc @rashmihunt

File: src/parser/xmlHandler.js [hunk 1]
```diff
@@ -249,7 +249,12 @@ class XMLHandler {
   }
 
   addAttributes(node, nsContext, descriptor, val, attrs) {
-    var elements = {}, attributes = {};
+    var attrDescriptors = (descriptor && descriptor.attributes) || [];
```
Comment (jannyHou) [inline src/parser/xmlHandler.js]: I may not have the knowledge to review the feature change in this repo, just the old code declares `element` as `{}` but the new code doesn't, if this is an intended change then never mind :)

File: test/request-response-samples/addList__qualified_attrs/response.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" >
+    <soapenv:Header/>
+    <soapenv:Body>
+        <ns1:addListResponse xmlns:ns1="urn:example" xmlns:ns2="urn:example" ns2:discriptor="d2">
+		    <ns1:record>r1</ns1:record>
+			<ns1:record>r2</ns1:record>
+		</ns1:addListResponse>
+    </soapenv:Body>
+</soapenv:Envelope>
```
Comment (b-admike) [inline test/request-response-samples/addList__qualified_attrs/response.xml]: newline here perhaps?

File: test/request-response-samples/addList__qualified_attrs/soap.messages.xsd [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <element name="addListResponse" type="msgs:AddListResponse"/>
+    	
+    <complexType name="AddListRequest">
+        <sequence>
+            <element name="record" type="string" minOccurs="1" maxOccurs="unbounded"/>
+        </sequence>
+        <attribute name="discriptor" type="string"/>
+    </complexType>
+    <element name="addList" type="msgs:AddListRequest"/>
+</schema>
```
Comment (b-admike) [inline test/request-response-samples/addList__qualified_attrs/soap.messages.xsd]: ditto

File: test/request-response-samples/addList__qualified_attrs/request.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "record": ["r1", "r2"],
+  "$attributes": {
+    "discriptor": "d1"
```
Comment (kjdelisle) [inline test/request-response-samples/addList__qualified_attrs/request.json]: Should be "descriptor", no?

**Pull request merged**